guides/source/layout.html.erb

```diff
@@ -39,59 +45,83 @@
         <li class="more-info"><a href="https://github.com/rails/rails">Contribute on GitHub</a></li>
       </ul>
     </div>
-  </div>
-  <div id="header">
+  </nav>
+  <header id="page_header">
     <div class="wrapper clearfix">
-      <div class="header-logo"><a href="index.html" title="Return to home page">Guides.rubyonrails.org</a></div>
-      <ul class="nav">
-        <li><a class="nav-item" href="index.html">Home</a></li>
-        <li class="guides-index guides-index-large">
-          <a href="index.html" id="guidesMenu" class="guides-index-item nav-item">Guides Index</a>
-          <div id="guides" class="clearfix" style="display: none;">
-            <hr />
-            <dl class="guides-section-container">
-              <% documents_by_section.each do |section| %>
-                <div class="guides-section">
-                  <dt><%= section['name'] %></dt>
+      <nav id="feature_nav"> <%# we use nav as an element to also provide the aria role and improve semantic meaning. %>
+        <div class="header-logo">
+          <a href="index.html" title="Return to home page">Guides</a>
+          <span id="version_switcher">
+            Version:
+            <select class="guides-version">
+              <option value="https://edgeguides.rubyonrails.org/" selected>Edge</option>
+              <option value="https://guides.rubyonrails.org/v7.1">7.1</option>
+              <option value="https://guides.rubyonrails.org/v7.0">7.0</option>
+              <option value="https://guides.rubyonrails.org/v6.1">6.1</option>
+              <option value="https://guides.rubyonrails.org/v6.0">6.0</option>
+              <option value="https://guides.rubyonrails.org/v5.2">5.2</option>
+              <option value="https://guides.rubyonrails.org/v5.1">5.1</option>
+              <option value="https://guides.rubyonrails.org/v5.0">5.0</option>
+              <option value="https://guides.rubyonrails.org/v4.2">4.2</option>
+              <option value="https://guides.rubyonrails.org/v4.1">4.1</option>
+              <option value="https://guides.rubyonrails.org/v4.0">4.0</option>
+              <option value="https://guides.rubyonrails.org/v3.2">3.2</option>
+              <option value="https://guides.rubyonrails.org/v3.1">3.1</option>
+              <option value="https://guides.rubyonrails.org/v3.0">3.0</option>
+              <option value="https://guides.rubyonrails.org/v2.3">2.3</option>
+            </select>
+          </span>
+        </div>
```

Comment:
I feel like the "Version: [ Edge ]" portion is a bit misaligned here, to a point that it bugs me just a little, but maybe it's just me... should these be aligned differently?

![Screenshot 2024-03-18 at 17 29 26](https://github.com/rails/rails/assets/26328/05832276-a23d-4f6d-a950-46ccdf551ebb)
![Screenshot 2024-03-18 at 17 30 09](https://github.com/rails/rails/assets/26328/262d3f3d-232c-4d20-a8b6-0feffd51770d)

Reply:
baseline is south by ~2 pixels. I'll get it to lock. Might just be the difference in fonts, but I'll have it work visually.

Reply:
<img width="337" alt="Screenshot 2024-03-18 at 9 13 32 PM" src="https://github.com/rails/rails/assets/2223/7136e7b6-b0b9-42bd-b977-076d57d0004a">

1px is all it took :)

Reply:
Resolved in 3b329554